base/services/w32time/W32Time.c

```diff
@@ -362,27 +363,43 @@ ServiceMain(DWORD argc, LPWSTR *argv)
     return; 
 }
 
-int wmain(int argc, WCHAR *argv[]) 
+
+
+BOOL WINAPI
+DllMain(HINSTANCE hinstDLL,
+        DWORD fdwReason,
+        LPVOID lpvReserved)
 {
-    SERVICE_TABLE_ENTRYW ServiceTable[] =
+    switch (fdwReason)
     {
-        {ServiceName, ServiceMain},
-        {NULL, NULL}
-    };
-
-    UNREFERENCED_PARAMETER(argc);
-    UNREFERENCED_PARAMETER(argv);
+        case DLL_PROCESS_ATTACH:
+            DisableThreadLibraryCalls(hinstDLL);
+            break;
 
-    DPRINT("W32Time: main starting.\n");
+        case DLL_PROCESS_DETACH:
+            break;
+    }
 
-    /* Start the control dispatcher thread for our service */
-    StartServiceCtrlDispatcherW(ServiceTable);
+    return TRUE;
+}
 
-    DPRINT("W32Time: main done.\n");
 
-    return 0; 
+HRESULT WINAPI W32TimeSyncNow(LPCWSTR cmdline, UINT blocking, UINT flags)
+{
+    int result;
+    result = SetTime();
+    if (result)
+    {
+        DPRINT("W32TimeSyncNow failed and clock not set.\n");
+        return 1;
```

Comment:
This is not an HRESULT.
If you want a 'generic' failure, use `E_FAIL`. (Although I would prefer to return a useful code)

Reply:
@learn-more, yes, this needs more work to enable the correct responses and I hope to eventually add this additional code. I would like to get this committed first and later improve on it. Thanks.